src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -95,6 +96,120 @@ public static FunctionPointerMethodSymbol CreateFromSource(FunctionPointerTypeSy
         public static FunctionPointerMethodSymbol CreateFromMetadata(CallingConvention callingConvention, ImmutableArray<ParamInfo<TypeSymbol>> retAndParamTypes)
             => new FunctionPointerMethodSymbol(callingConvention, retAndParamTypes);
 
+        public FunctionPointerMethodSymbol SubstiteParameterSymbols(TypeWithAnnotations substitutedReturnType, ImmutableArray<TypeWithAnnotations> substitutedParameterTypes)
+            => new FunctionPointerMethodSymbol(
+                this.CallingConvention,
+                this.RefKind,
+                substitutedReturnType,
+                this.RefCustomModifiers,
+                this.Parameters,
+                substitutedParameterTypes);
+
+        internal FunctionPointerMethodSymbol MergeEquivalentTypes(FunctionPointerMethodSymbol signature, VarianceKind variance)
+        {
+            var mergedParameterTypes = ImmutableArray<TypeWithAnnotations>.Empty;
+            bool hasParamChanges = false;
+            if (_parameters.Length > 0)
+            {
+                var builder = ArrayBuilder<TypeWithAnnotations>.GetInstance(_parameters.Length);
+                var paramVariance = variance switch
+                {
+                    VarianceKind.In => VarianceKind.Out,
+                    VarianceKind.Out => VarianceKind.In,
+                    _ => VarianceKind.None,
+                };
+
+                for (int i = 0; i < _parameters.Length; i++)
+                {
+                    var mergedParameterType = _parameters[i].TypeWithAnnotations.MergeEquivalentTypes(signature._parameters[i].TypeWithAnnotations, paramVariance);
+                    builder.Add(mergedParameterType);
+                    if (!mergedParameterType.IsSameAs(_parameters[i].TypeWithAnnotations))
+                    {
+                        hasParamChanges = true;
+                    }
+                }
+
+                mergedParameterTypes = builder.ToImmutableAndFree();
+            }
+
+            var mergedReturnType = ReturnTypeWithAnnotations.MergeEquivalentTypes(signature.ReturnTypeWithAnnotations, variance);
+            if (hasParamChanges || !mergedReturnType.IsSameAs(ReturnTypeWithAnnotations))
+            {
+                return SubstiteParameterSymbols(mergedReturnType, mergedParameterTypes);
+            }
+            else
+            {
+                return this;
+            }
+        }
+
+        public FunctionPointerMethodSymbol SetNullabilityForReferenceTypes(Func<TypeWithAnnotations, TypeWithAnnotations> transform)
```

Comment:
>SetNullabilityForReferenceTypes [](start = 43, length = 31)

I assume somewhere we have a function that performs the opposite operation, i.e. builds the transform flags. Could you point me to it?

Reply:
It's not currently implemented, see `FunctionPointerTypeSymbol.AddNullableTransforms` (line 111 in this PR)

---
In reply to: [406857994](https://github.com/dotnet/roslyn/pull/43041#discussion_r406857994) [](ancestors = 406857994)